### ISSUE TITLE
Round materialish buttons are useless and hide important text underneath

### ISSUE THREAD

**Issue by dllud** [rdoeffinger/Dictionary#50]
First of all thanks @rdoeffinger for keeping this project alive!

I find that the round materialish buttons you recently introduced in the app are useless and actually hinder usability instead of improving it.

Useless because all their actions were already 0 or 1 clicks away in the old interface:
- **Searching** is the default action when you enter a dictionary. The cursor is automatically in the search box and keyboard is opened. Thus we need zero clicks to get to it. Even if we clicked somewhere else, searching again required only 1 click in the search box, which is a big target that you can easily hit with your fingers.
- **Clearing** the current search word took 1 click in the `X` button to the right of the search box. Again this is a big target that no one misses.
- **Switching languages** also took a single click in the flag, again a big target. This one was somewhat hidden. It is hard for a new user to guess that clicking the flag changes languages. The straightforward solution would be to place (watermark) the two arrows symbol on top of the flag.

Summing up: these new buttons solve no usability issue.

Now, even worse, is the fact that they actually are a usability hindrance, because they hide important text underneath. Checkout what happens in my device:

![quickdic-material-buttons](https://cloud.githubusercontent.com/assets/1716156/19929022/2edb302a-a0f9-11e6-853d-371e41c666b1.png)

90% of the time we use Quickdic with the keyboard opened. These new buttons take a lot of the precious little space we have left. With them, I am forced to hit the back button to hide the keyboard in order the read the translations. That's something I didn't had to do before. Thus they actually increased the number of clicks for me.

Taking this into consideration, I would urge you to remove these buttons, or at least add a preference's option to do so.

### Tips for future UI improvements

The original UI by Thad Hughes is actually really good. All major actions are easy to do with it. I would exercise caution before adding modifications.

I see that you are in the mood to update it to a full blown Material UI. If so, please consider starting such work in a new branch. Besides avoiding breaking the UI on the production app, a new branch would allow you to start from scratch. It is rarely a good move to work iteratively on top of an old UI design. Better start anew and see whether applying Material solutions to this problem renders a better UI than the original Holo theme.

**Comment by rdoeffinger** [comment]
Thanks for your feedback and I am happy to recruit your help to find a better solution.
But it seems to me you are missing or misunderstanding some things around this change:
- I am not changing the UI for fun or to follow some guidelines/themes, but I am reacting to specific user issues and requests
- In this specific case, a button on the top left is a pain to use on any large phone (even the search field is problematic to reach), see e.g. issue #48 
- Personally, I do use smaller devices for testing, but I found it quite unusable with the keyboard open even before. One of the reasons why I changed the enter/"done" button on the keyboard to hide it, which I found more comfortable than pressing "back". Using it that way, it seemed to work fine even with the extra buttons, but if you usually leave the keyboard open that does change things.
- It does solve some minor issues like that finding your way back to the dictionary selection was quite a pain, but that is just a minor side-effect.
(I also have to admit I found your criticism a bit harsh, considering that more space is wasted on your screenshot by an oversized action bar - on my phone its relative size is a good bit smaller - and a really useless empty suggestion bar above the keyboard, sorry if it shows a bit in my answer :) )

Now, as to what to do about it, I have come up with the following options, but I like none of them.
If you convince me that something is the perfect solution I'll do my best to do it.
- reduce the size of the buttons (and/or margins). Very much a compromise that I suspect leaves nobody happy
- add an option. The chicken-out solution. It works as a band-aid, but adds complexity, new users probably will not be aware of it and I'll probably end up testing only one of the settings most of the time...
- make it dependent on screen size. I don't much like the idea that someone e.g. switching their phone will suddenly be faced with a different UI though.
- hide the buttons when the keyboard is visible. Annoying for people with large screens as the keyboard shows by default and we're back to the original problem. Also, I haven't found a way to do this reliably.
- as above, but instead do it as soon as the search field contains text
- possible compromise mixture of the above: force the action buttons to always leave a certain minimum area free (i.e. do not allow them to move up all the way). Not sure how to do that, and might look ugly.
- remove only the search button. When I tried it, it didn't really improve things much whether just half or all of the text on the right side is obscured...

As to the suggestion of using a different branch, as mentioned I do not plan any such large changes.
Plus a branch would not fix that it would only be me that tries it out (and I concluded that this change should be an overall improvement), and it would just mean that I get the complaints about the changes later. Also in case of serious issues it is possible to get any old version from F-droid or here, though I admit that Google Play does unfortunately not offer that option.

**Comment by dllud** [comment]
Thanks for the thoughtful explanation. Now I see where this is coming from, I had never used QuickDic in a tablet or any other large screen device.

> more space is wasted on your screenshot by an oversized action bar

Agreed. The action bar could be thinner. It probably has to do with the low pixel density of this device, a Samsung GT-i9100. It seems that QuickDic has a fixed height action bar or doesn't adjust it correctly for a hdpi screen. Other apps seem to do that adjustment or have a thinner action bar to start with, e.g. [Conversations](https://f-droid.org/repository/browse/?fdid=eu.siacs.conversations) and [FreeOTP](https://f-droid.org/repository/browse/?fdid=org.fedorahosted.freeotp).
However, not all that height was taken by the action bar. Some of it was taken by the box surrounding the matched word. Take a look at the screenshot bellow for a clear example. As you can see the box that surrounds each matched word has an oversized top padding, that is much larger that the bottom padding. 

> and a really useless empty suggestion bar above the keyboard

That one is actually really useful. Unfortunately the previous screenshot doesn't show it in use. Here is what actually happens when I am typing in QuickDic (I was trying to type `correction`):
![quickdic-searching](https://cloud.githubusercontent.com/assets/1716156/19965943/54675080-a1c1-11e6-9d43-2fe9e3d627f3.png)


Solving the above issues of oversized action bar and word box would help, but wouldn't solve the entire problem.

Going through all your solution proposals leaves me with no perfect candidate. The solution I am most fond of would be to hide the buttons, or even better, make them semi-transparent, when the keyboard is open *and* the available heigh is smaller than a given value.
This would fulfil your requirements: no button size compromise, no opt-out, same UI across all devices, etc. Though, it might be somewhat hard to implement. As such, I would leave things as they are, and see if anyone else besides me complains about this. Perhaps in the meantime I'll get used to closing the keyboard everytime I type.

**Comment by rdoeffinger** [comment]
Well, I am not entirely giving up on this.
I just had at look at what this looks like on one of the platforms I usually test on, and I confirmed that somehow the UI proportions are completely different, so that must be part of why I saw less of a difference:
![screenshot_20161106_170746](https://cloud.githubusercontent.com/assets/5464937/20038919/d36ba760-a43b-11e6-8782-997632dda724.png)
So on some phones the action buttons are shown much smaller already. Thus I might as well go for consistency by making them smaller everywhere, it looks fairly good even on the big phones IMHO, even though it is against the material guidelines.
Btw I do of course know about the suggestions bar on the keyboard, but e.g. on that system it at least disappears when it has no suggestions to show...
I also forgot to mention: the reason to have the search button that changes into X (clear) instead of having it disappear is because I'd like to get rid of the "x" in the search field. On smaller devices that text field is rather too small, and the extra space would be nice to have there. Haven't figured out a good way to do it yet though. If you think it's a really bad idea you can say it right now before I do it ;)

**Comment by dllud** [comment]
> So on some phones the action buttons are shown much smaller already. Thus I might as well go for consistency by making them smaller everywhere,

That would help somehow. I think it would let me read the 3rd entry, probably reducing the number of times I have to close the keyboard.

The search button turning into `X` is quite useful, I have been using it often. If you keep it like that, I see no problem in dropping the `X` in the search field.

The one I rarely use is the switch language button. When I am reading or writing a text in a given language, I keep searching for words in that language. Only an hour or so later, when I grab a text in some other language, do I need to switch languages in QuickDic.
Thus, with my usage pattern, moving my finger to click the flag is no big deal, even in a large screen, because I do it only once in a while. This seems to be the exact opposite of other users as pointed out by @moshpirit in #48.

For me, in a large screen, it would be more useful to have the navigation buttons (up ▲ and down ▼) down there in round buttons. Of course this would totally spoil the UI in a small screen. Just letting you know so you have some insights into how people use the app.

**Comment by rdoeffinger** [comment]
Thanks everyone for the comments. I'll close this for now, on the assumption that current behaviour is a reasonable compromise with no obvious ways to improve, and thus expect no further work on it.

**Comment by dllud** [comment]
>  current behaviour is a reasonable compromise with no obvious ways to improve

You're right.

Still, could you make the buttons smaller everywhere like you proposed above? That would help, at least in my case.

**Comment by rdoeffinger** [comment]
Hm, I did that already for 5.3.1 (ID 78). Are they not smaller on your device?

**Comment by dllud** [comment]
That version still didn't make it into F-Droid. That's why I missed it, sorry.
I'll report back as soon as it enters into F-Droid.

**Comment by dllud** [comment]
I've just tried the new version with the smaller buttons. It does help a lot. Now I can read the 3rd entry and sometimes the 4th one. Thanks!
